### PR TITLE
feat: add python3 tool implementation

### PR DESCRIPTION
**Pull request by jherr-dev**
Implements a tool to run python3 scripts via Toolchest. Script must use the expected local input and output directories but are otherwise not limited.

### CODE REVIEW

File: toolchest_client/tools/api.py [hunk 1]
```diff
@@ -388,6 +391,44 @@ def megahit(output_path=None, tool_args="", read_one=None, read_two=None, interl
     return output
 
 
+def python3(script, inputs=[], output_path=None, tool_args="", **kwargs):
+    """Runs Python3 via Toolchest.
+    NOTE: This tool is currently only available for customers with contracts
```
Comment (lebovic) [inline toolchest_client/tools/api.py:396]: Rephase: e.g. "this is a restricted tool, you can request access at ___"

File: toolchest_client/tools/api.py [hunk 2]
```diff
@@ -388,6 +391,44 @@ def megahit(output_path=None, tool_args="", read_one=None, read_two=None, interl
...
+    The script has a few requirements if inputs or outputs are expected. Inputs will be uploaded to `./input/` so the
+    script must use that prefix with hardcoded paths or when adding to tool_args to be parsed via sys.argv. Any output
+    that is expected to be returned needs to be stored at `./output/`. The directory will be compressed via pigz and
+    decompressed on the client.
```
Comment (lebovic) [inline toolchest_client/tools/api.py:400]: This can be rephrased to be more digestible

File: toolchest_client/tools/api.py [hunk 3]
```diff
@@ -388,6 +391,44 @@ def megahit(output_path=None, tool_args="", read_one=None, read_two=None, interl
...
+
+    :param script: Path to the python script to run
+    :param inputs: (optional) Path(s) to the input files to run with the python script. Will be uploaded to a directory
+    `./input/` for use in the on the instance.
+    :param output_path: (optional) Local path to where the output file(s) will be downloaded.
+    :param tool_args: (optional) Additional arguments to be passed to Python3. Should NOT specify the script here but
+    any other args after the script should be included.
```
Comment (lebovic) [inline toolchest_client/tools/api.py:407]: Rephrase with a positive example and/or a more clear example of what not to do, or just remove the added context.

(More specifically, modify or remove "Should NOT specify the script here but any other args after the script should be included.")

File: toolchest_client/tools/api.py (hunk 2 above)
Comment (lebovic) [reply toolchest_client/tools/api.py:400]: "The script has a few requirements if inputs or outputs are expected." – not necessary

"Inputs will be uploaded to `./input/` so the script must use that prefix with hardcoded paths or when adding to tool_args to be parsed via sys.argv" - hard to understand

"Any output that is expected to be returned needs to be stored at `./output/`" - make more clear what this means, what needs to store what at ./output/?

"The directory will be compressed via pigz and decompressed on the client." - more detail than is necessary, but still lacks implication (maybe "The output directory is compressed and returned, meaning chaining Python 3 calls via Toolchest is not possible")

File: toolchest_client/tools/api.py [hunk 4]
```diff
@@ -388,6 +391,44 @@ def megahit(output_path=None, tool_args="", read_one=None, read_two=None, interl
...
+        >>> toolchest.python3(
+        ...     script="./path/to/script.py",
+        ...     output_path="./path/to/output/base",  # outputs
+        ...     tool_args="",
+        ... )
+    """
+    if type(inputs) is str:
+        inputs = [inputs]
+    elif type(inputs) is not list:
+        raise ToolchestException('Unknown input type')
```
Comment (lebovic) [inline toolchest_client/tools/api.py:419]: Why was this added, did you hit an error? I'm curious because this seems like a new pattern

File: toolchest_client/tools/python3.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    """
+    def __init__(self, tool_args, inputs, output_path, **kwargs):
+        super().__init__(
+            tool_name="python3",
+            tool_version="3.9.1",
+            tool_args=tool_args,
+            output_path=output_path,
+            inputs=inputs,
+            min_inputs=1,
+            max_inputs=100,
```
Comment (lebovic) [inline toolchest_client/tools/python3.py:24]: This is default behavior, safe to remove?

File: toolchest_client/tools/python3.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+            parallel_enabled=False,
```
Comment (lebovic) [inline toolchest_client/tools/python3.py:25]: Same here

File: toolchest_client/tools/tool_args.py [hunk 7]
```diff
@@ -385,6 +385,11 @@
             "--min-contig-len": 1,
         },
     },
+    "python3": {
```
Comment (lebovic) [inline toolchest_client/tools/tool_args.py:388]: Might be worth blacklisting some args – e.g. `-i` for spawning an interactive Python environment

**Review by lebovic — CHANGES_REQUESTED** [review]
Overall looks good, comments center around documentation

File: toolchest_client/tools/api.py (hunk 4 above)
Comment (jherr-dev) [reply toolchest_client/tools/api.py:419]: Just being overly cautious. I'll remove it

**Review by lebovic — APPROVED** [review]
